core/src/test/java/io/opentelemetry/android/export/BufferDelegatingSpanExporter.kt

```diff
@@ -0,0 +1,116 @@
+import io.opentelemetry.sdk.testing.exporter.InMemorySpanExporter
+import io.opentelemetry.sdk.trace.data.SpanData
+import org.junit.Test
+
+class BufferDelegatingSpanExporterTest {
+    @Test
+    fun `test setDelegate`() {
+        val bufferDelegatingSpanExporter = BufferDelegatingSpanExporter()
+        val spanExporter = InMemorySpanExporter.create()
+
+        val spanData: SpanData = mockk<SpanData>()
+        bufferDelegatingSpanExporter.export(listOf(spanData))
+        bufferDelegatingSpanExporter.setDelegate(spanExporter)
+
+        assertThat(spanExporter.finishedSpanItems)
+            .containsExactly(spanData)
+    }
+
+    @Test
+    fun `test buffer limit handling`() {
+        val bufferDelegatingSpanExporter = BufferDelegatingSpanExporter(10)
+        val spanExporter = InMemorySpanExporter.create()
+
+        for (i in 1..11) {
+            val spanData: SpanData = mockk<SpanData>()
+            bufferDelegatingSpanExporter.export(listOf(spanData))
+        }
+
+        bufferDelegatingSpanExporter.setDelegate(spanExporter)
+
+        assertThat(spanExporter.finishedSpanItems)
+            .hasSize(10)
+    }
+
+    @Test
+    fun `test flush with delegate`() {
+        val bufferDelegatingSpanExporter = BufferDelegatingSpanExporter()
+        val delegate = spyk<InMemorySpanExporter>()
+
+        val spanData: SpanData = mockk<SpanData>()
+        bufferDelegatingSpanExporter.export(listOf(spanData))
+
+        bufferDelegatingSpanExporter.setDelegate(delegate)
+
+        bufferDelegatingSpanExporter.flush()
+
+        verify { delegate.flush() }
+    }
+
+    @Test
+    fun `test export with delegate`() {
+        val bufferDelegatingSpanExporter = BufferDelegatingSpanExporter()
+        val delegate = spyk<InMemorySpanExporter>()
+
+        val spanData: SpanData = mockk<SpanData>()
+        bufferDelegatingSpanExporter.export(listOf(spanData))
+
+        verify(exactly = 0) { delegate.export(any()) }
+
+        bufferDelegatingSpanExporter.setDelegate(delegate)
+
+        verify(exactly = 1) { delegate.export(any()) }
+
+        val spanData2: SpanData = mockk<SpanData>()
+        bufferDelegatingSpanExporter.export(listOf(spanData2))
+
+        verify(exactly = 2) { delegate.export(any()) }
+    }
+
+    @Test
+    fun `test shutdown with delegate`() {
+        val bufferDelegatingSpanExporter = BufferDelegatingSpanExporter()
+        val delegate = spyk<InMemorySpanExporter>()
+
+        bufferDelegatingSpanExporter.setDelegate(delegate)
+
+        bufferDelegatingSpanExporter.shutdown()
+
+        verify { delegate.shutdown() }
+    }
+
+    @Test
+    fun `test flush without delegate`() {
+        val bufferDelegatingSpanExporter = BufferDelegatingSpanExporter()
+
+        val spanData: SpanData = mockk<SpanData>()
+        bufferDelegatingSpanExporter.export(listOf(spanData))
+
+        val flushResult = bufferDelegatingSpanExporter.flush()
+
+        assertThat(flushResult.isSuccess).isTrue()
```

Comment:
This gives a much nicer error message:
```suggestion
        assertThat(flushResult).prop(CompletableResultCode::isSuccess).isTrue()
```

Reply:
they use assertJ not assertK. I could use `extracting`. I just set it up and caused failures, the error messages seemed the same to me: 

1. `assertThat(flushResult.isSuccess).isFalse()`
<img width="1232" alt="Screenshot 2024-11-11 at 12 56 10 PM" src="https://github.com/user-attachments/assets/931421ed-ffd1-4924-a342-054e7fcc6225">


2. `assertThat(flushResult).extracting(Com
<img width="1259" alt="Screenshot 2024-11-11 at 12 56 31 PM" src="https://github.com/user-attachments/assets/85f35bb1-a3a2-4022-93fa-112556d60900">
pletableResultCode::isSuccess).isSameAs(false)`

Reply:
im going to resolve this, but if there is further discussion I will open i t back up :)